sql/src/test/java/io/crate/execution/dml/upsert/CheckConstraintsTest.java

```diff
@@ -0,0 +1,125 @@
+import io.crate.metadata.doc.DocTableInfo;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.LinkedHashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Testing {@linkplain io.crate.execution.dml.upsert.CheckConstraints},
+ * as well as various cases related to:
+ *
+ * <pre>
+ *     CONSTRAINT &lt;name&gt; CHECK &lt;boolean expression&gt;
+ * </pre>
+ */
+public class CheckConstraintsTest extends CrateDummyClusterServiceUnitTest {
+
+    private SQLExecutor sqlExecutor;
+    private DocTableInfo docTableInfo;
+    private CheckConstraints checkConstraints;
+    private TransactionContext txnCtx;
+
+    @Before
+    public void setUpExecutor() throws Exception {
+        sqlExecutor = SQLExecutor.builder(clusterService)
+            .addTable("CREATE TABLE t (" +
+                      "    id int," +
+                      "    qty int," +
+                      "    sentinel boolean CONSTRAINT sentinel CHECK(sentinel)," +
+                      "    CONSTRAINT id_is_even CHECK(id % 2 = 0))")
+            .build();
+        docTableInfo = sqlExecutor.resolveTableInfo("t");
+        txnCtx = CoordinatorTxnCtx.systemTransactionContext();
+        checkConstraints = new CheckConstraints(
+            txnCtx,
+            new InputFactory(sqlExecutor.functions()),
+            FromSourceRefResolver.WITHOUT_PARTITIONED_BY_REFS,
+            docTableInfo);
+    }
+
+    @Test
+    public void test_validate_fails_when_check_expr_is_false() throws Exception {
+        expectedException.expectMessage(
+            "Failed CONSTRAINT sentinel CHECK (\"sentinel\") and values {id=280278, qty=42, sentinel=false}");
+        checkConstraints.validate(mapOf(
+            "id", 280278,
+            "qty", 42,
+            "sentinel", false));
+    }
+
+    @Test
+    public void test_validate_succeeds_when_check_expr_is_true() throws Exception {
+        checkConstraints.validate(mapOf(
+            "id", 280278,
+            "qty", 42,
+            "sentinel", true));
+    }
+
+    @Test
+    public void test_validate_succeeds_when_check_expr_is_null() throws Exception {
+        checkConstraints.validate(mapOf(
+            "id", 280278,
+            "qty", 42,
+            "sentinel", null));
+    }
+
+    @Test
+    public void test_cannot_have_two_check_constraints_of_same_name() throws Exception {
+        expectedException.expectMessage(
+            "a check constraint of the same name is already declared [id_is_even]");
+        SQLExecutor.builder(clusterService)
+            .addTable("CREATE TABLE t (" +
+                      "    id int CONSTRAINT id_is_even CHECK(id % 2 = 0)," +
+                      "    qty int," +
+                      "    CONSTRAINT id_is_even CHECK(id % 2 = 0))")
+            .build();
+    }
+
+    private static Map<String, Object> mapOf(Object... items) {
```

Comment:
This looks like `Map.of` or, if you need stable ordering, `ImmutableMap.of` ?

Reply:
Map.of did not let me have a null value